userlevel/dpdk.mk

```diff
@@ -73,7 +73,9 @@ ifeq ($(CONFIG_RTE_LIBRTE_VHOST_USER),n)
 _LDLIBS-$(CONFIG_RTE_LIBRTE_VHOST)          += -lfuse
 endif
 
+ifeq ($(HAVE_ZLIB),y)
 _LDLIBS-$(CONFIG_RTE_LIBRTE_BNX2X_PMD)      += -lz
+endif
```

Comment:
This will add `-lz` multiple times. I think the better approach would be something like

“add -lz iff at least one of `CONFIG_RTE_LIBRTE_BNX2X_PMD` and `CONFIG_RTE_LIBRTE_QEDE_PMD` is enabled“…presumably those config options will NOT be enabled if `-lz` is absent.

The problem you might have been solving is that you had `CONFIG_RTE_LIBRTE_QEDE_PMD` but not `…BNX2X_PMD`, but the configury here only added `-lz` if `BNX2X_PMD`.

Can you confirm? A smaller change that built on the approach in the old makefile (i.e. config dependency) would be better.